text/0004-iso-8601.md

```diff
@@ -0,0 +1,51 @@
+- Start Date: 2023-10-06
+- Contributors:
+  - [Zak Burke](zburke@ebsco.com)
+- RFC PRs:
+  - DRAFT REVIEW: https://github.com/folio-org/rfcs/pull/13
+  - PRELIMINARY REVIEW: TBD
+  - PUBLIC REVIEW: TBD
+  - FINAL REVIEW: TBD
+- Outcome: <!-- Leave this blank. Will eventually be either ACCEPTED or REJECTED -->
+
+# Date-time values must comply with IETF RFC-3339
+
+## Summary
+
+* Validation of date-time values must assert compliance with IETF RFC-3339.
+* Date-time values in storage must be compliant with IETF RFC-3339 Section 5.6.
+* APIs must respect date-time values when retrieving a range of values by date-time in a query.
+
+## Motivation
+
+Compliance with [IETF RFC-3339](https://www.rfc-editor.org/rfc/rfc3339.html) as stated in our ["Working with dates and times" documentation](https://dev.folio.org/guides/dates-and-times/).
+
+## Detailed Explanation/Design
+
+Code that validates date-time values must do so in accordance with [IETF RFC-3339 Section 5.6](https://www.rfc-editor.org/rfc/rfc3339.html#section-5.6).
+
+Date-time values at rest must be fully compliant with [IETF RFC-3339 Section 5.6](https://www.rfc-editor.org/rfc/rfc3339.html#section-5.6).
+
+Code that handles date-time values must respect all aspects of such values, including but not limited to their `time-offset` attributes. For example, an API that retrieves entries in a given date-time range must return the same results given any of the following arguments:
+
+* `2023-10-05T12:51:12.000-04:00` and `2023-10-06T12:51:12.000-04:00`
+* `2023-10-05T16:51:12.000Z` and `2023-10-06T16:51:12.000Z`
+* `2023-10-05T11:51:12.000-05:00` and `2023-10-06T18:51:12.000+02:00`
+
+### Related Jiras
+
+* [MODAUD-172](https://issues.folio.org/browse/MODAUD-172)
+
+## Risks and Drawbacks
+
+Existing documentation states that, "The APIs assume UTC on input, and store and return timestamps in UTC". Thus, APIs may assume date-time values without a `time-offset` attributes are UTC, which may be incorrect given that ISO-8601 states that [when no UTC offset is provided, "the time is assumed to be in local time"](https://en.wikipedia.org/wiki/ISO_8601#Local_time_(unqualified)).
```

Comment:
> APIs that allow a time-offset can require the time-offset to be zero

APIs that require a time-offset to be zero do not, in fact, allow a time-offset.

Reply:
If we pursue this implementation, then values with non-zero offsets must be rejected. Under this scenario, MODAUD-172 is still a legit bug but the appropriate fix is to return a 400 instead of a 200 with the correct content. I think that's a worse fix, but I also think it's better than the current implementation that sends an incorrect response.